src/script/lua_api/l_env.cpp

```diff
@@ -609,6 +609,12 @@ int ModApiEnvMod::l_find_node_near(lua_State *L)
 		return 0;
 	}
 
+	// Client API limitations
+	if (getClient(L) &&
+			getClient(L)->getCSMFlavourLimits() & CSMFlavourLimit::CSM_FL_LOOKUP_NODES) {
+		// TODO
```

Comment:
line of sight is not this call. Each call has is own role. We can add line_in_sight call and it's easy to limit it to not show nodes behind a solid node